### PR TITLE
refactor(checkbox): move out of experimental folder

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour

- Move `Checkbox` and `CheckboxGroup` into `src/components`.
- Update Cypress tests now `Checkbox` is in the Design System Storybook folder
- Replace `react-test-renderer` tests with `enzyme` ones

### Current behaviour
These components are currently in the experimental folder.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/checkbox/checkbox-group.spec.js [hunk 1]
```diff
@@ -35,10 +34,6 @@ function render(props, childProps, renderer = mount) {
 }
 
 describe("CheckboxGroup", () => {
-  it("renders as expected", () => {
-    expect(render({}, {}, TestRenderer.create)).toMatchSnapshot();
-  });
-
```
Comment (grabkowski) [inline src/components/checkbox/checkbox-group.spec.js:41]: Do we no longer need that snapshot? I think that you need at least to write a test that checks if the structure is rendered as expected.

**Review by ianforster-sage — APPROVED** [review]

File: src/components/checkbox/checkbox-group.spec.js (hunk 1 above)
Comment (sjohnson-sage) [reply src/components/checkbox/checkbox-group.spec.js:41]: What value would that provide that Chromatic doesn't? We lost no test coverage by removing this as well

**Review by grabkowski — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Comment by sjohnson-sage** [comment]
Closing as the CI is completely broken, I believe because I opened the PR with the target branch as master, and later changed it once I realised my mistake

**Review by ianforster-sage — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**